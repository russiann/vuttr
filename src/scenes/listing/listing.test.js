import React from 'react';
import {render, cleanup} from '@testing-library/react';
import configureStore from 'redux-mock-store';
import Tools from './listing.scene';
import enhancer from './listing.enhancer';
import {Provider} from 'react-redux';
import Immutable from 'seamless-immutable';

const mockStore = configureStore();

const mockRematchStore = (initialState, reducersAndEffects = {}) => {
  const store = mockStore(initialState);
  Object.assign(store.dispatch, reducersAndEffects);
  return store;
};

/**
|--------------------------------------------------
| Mocks
|--------------------------------------------------
*/

jest.mock('./components/tool-item/tool-item.component', () => props => (
  <div data-testid="tool-item" />
));
jest.mock(
  '../../common/components/searchfield/searchfield.component',
  () => props => <div data-testid="search-field" />
);
jest.mock(
  '../../common/components/checkbox/checkbox.component',
  () => props => <input data-testid="checkbox" {...props} />
);
jest.mock('../../common/components/button/button.component', () => props => (
  <button data-testid="Button" {...props} />
));
jest.mock('../../common/components/modal/modal.component', () => () => (
  <div data-testid="modal" />
));

/**
|--------------------------------------------------
| Tests
|--------------------------------------------------
*/

afterEach(cleanup);

describe('enhancer', () => {
  it('should run find when mount', () => {
    const EmptyComponent = () => null;
    const reduxStoreMock = {
      listing: Immutable({
        newToolModalOpened: false,
        confirmModal: {}
      }),
      tools: Immutable({
        data: [],
        filters: {}
      })
    };

    const dispatch = {
      listing: {},
      tools: {
        find: jest.fn()
      }
    };

    const store = mockRematchStore(reduxStoreMock, dispatch);

    const EnhancedComponent = enhancer(EmptyComponent);
    render(
      <Provider store={store}>
        <EnhancedComponent />
      </Provider>
    );
    expect(dispatch.tools.find).toHaveBeenCalled();
  });
});

describe('scene', () => {
  it('snapshot', () => {
    const {asFragment} = render(<Tools />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show tools', () => {
    const propsMock = {
      data: [
        {
          id: 3,
          title: 'fastify',
          link: 'https://www.fastify.io/',
          description:
            'Extremely fast and simple, low-overhead web framework for NodeJS. Supports HTTP2.',
          tags: ['a', 'b', 'c']
        },
        {
          title: 'test-tool',
          description: 'test-tool description wow',
          tags: ['d', 'e', 'f'],
          id: 5
        },
        {
          title: 'Feathers.js',
          description:
            'A REST and real-time API layer for Node.js, React Native and the browser.',
          link: 'https://feathersjs.com/',
          tags: ['g', 'h'],
          id: 6
        }
      ]
    };
    const {asFragment, queryAllByTestId} = render(<Tools {...propsMock} />);
    const tags = queryAllByTestId('tool-item');
    expect(asFragment()).toMatchSnapshot();
    expect(tags).toHaveLength(3);
  });
});
