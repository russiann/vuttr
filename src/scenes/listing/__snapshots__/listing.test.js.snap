// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scene should show tools 1`] = `
<DocumentFragment>
  .emotion-10 {
  max-width: 960px;
  margin: 0 auto;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  padding-top: 15px;
  padding-bottom: 15px;
}

.emotion-0 {
  width: 40px;
  height: 40px;
  border: 2px solid #ff6007;
  border-radius: 2px;
  margin-right: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  font-size: 23px;
  -webkit-letter-spacing: 6px;
  -moz-letter-spacing: 6px;
  -ms-letter-spacing: 6px;
  letter-spacing: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-6 emotion-7"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 3h6a4 4 0 0 1 4 4v14a3 3 0 0 0-3-3H2z"
          />
          <path
            d="M22 3h-6a4 4 0 0 0-4 4v14a3 3 0 0 1 3-3h7z"
          />
        </svg>
      </div>
      <div>
        <div
          class="emotion-2 emotion-3"
        >
          VUTTR
        </div>
        <div
          class="emotion-4 emotion-5"
        >
          Very Useful Tools to Remember
        </div>
      </div>
    </div>
    <div
      class="emotion-8 emotion-9"
    >
      <div
        data-testid="search-field"
      />
      <input
        css="[object Object]"
        data-testid="checkbox"
        label="Search in tags only"
      />
      <button
        data-testid="Button"
      >
         add 
      </button>
    </div>
    <div
      data-testid="tool-item"
    />
    <div
      data-testid="tool-item"
    />
    <div
      data-testid="tool-item"
    />
    <div
      data-testid="modal"
    />
    <div
      data-testid="modal"
    />
  </div>
</DocumentFragment>
`;

exports[`scene snapshot 1`] = `
<DocumentFragment>
  .emotion-10 {
  max-width: 960px;
  margin: 0 auto;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  padding-top: 15px;
  padding-bottom: 15px;
}

.emotion-0 {
  width: 40px;
  height: 40px;
  border: 2px solid #ff6007;
  border-radius: 2px;
  margin-right: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  font-size: 23px;
  -webkit-letter-spacing: 6px;
  -moz-letter-spacing: 6px;
  -ms-letter-spacing: 6px;
  letter-spacing: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-6 emotion-7"
    >
      <div
        class="emotion-0 emotion-1"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 3h6a4 4 0 0 1 4 4v14a3 3 0 0 0-3-3H2z"
          />
          <path
            d="M22 3h-6a4 4 0 0 0-4 4v14a3 3 0 0 1 3-3h7z"
          />
        </svg>
      </div>
      <div>
        <div
          class="emotion-2 emotion-3"
        >
          VUTTR
        </div>
        <div
          class="emotion-4 emotion-5"
        >
          Very Useful Tools to Remember
        </div>
      </div>
    </div>
    <div
      class="emotion-8 emotion-9"
    >
      <div
        data-testid="search-field"
      />
      <input
        css="[object Object]"
        data-testid="checkbox"
        label="Search in tags only"
      />
      <button
        data-testid="Button"
      >
         add 
      </button>
    </div>
    <div
      data-testid="modal"
    />
    <div
      data-testid="modal"
    />
  </div>
</DocumentFragment>
`;
